### PR TITLE
Community page

### PR DESCRIPTION
**Pull request by iAdramelk**
~~Strongly under development. PR only to check on Heroku.~~

Purpose:
- Provide an overview of what's going on with the project on the community side.
- For everyone who is meeting DVC for the first time.
- Help ppl find all the resources that might be useful to community members in one place.
- Emphasize that we are supporting "ambasssadors" among community members: ppl who give talks, want to organize meetup, etc.
- Remove some of the scattered mentions/links to community resources – which exact ones? We'll see.

Motivation:
- Most of the community page content was already there but scattered over the website and sometimes hard to find: "resources, channels of communication being pretty disorganized and mentioned both in the top menu, in the "support" section, in the "help" section, etc."

> From https://github.com/iterative/dvc.org/pull/874#issuecomment-570306262.

### CODE REVIEW

**Comment by jorgeorpinel** [comment]
Still strongly under development?

**Comment by shcheklein** [comment]
@jorgeorpinel it's definitely ready for the initial review.

**Comment by iAdramelk** [comment]
@jorgeorpinel it's functionally complete, but visual side needs additional work (mobile version, hover/focus/active states, cross browser testing) and I want to refactor the code a little.

But yes, we can start reviewing the code now already.

File: src/Community/Block/index.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import { Action, Content, Icon, Header, Wrapper } from './styles'
```
Comment (jorgeorpinel) [inline src/Community/Block/index.js]: I wouldn't call the styled container file `styles.js` though, sounds like we're talking about plain CSS or maybe Sass, etc. (especially when importing `from './styles'`.

How about `styled.js`? Or maybe even `styled_comps.js`. Small change, I know, but meaningful:

```suggestion
import { Action, Content, Icon, Header, Wrapper } from './styled'
```

File: src/Community/Block/index.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+export default function Block({ action, children, icon, large, title }) {
+  const hasAction = !!action
```
Comment (jorgeorpinel) [inline src/Community/Block/index.js]: Random q for @iAdramelk: Do you usually prefer to use function components in React/Next (instead of classes)? Apparently they officially prefer them... See https://reactjs.org/docs/hooks-intro.html

File: src/Community/Block/index.js (hunk 1 above)
Comment (jorgeorpinel) [reply src/Community/Block/index.js]: > Or maybe even `styled_comps.js`

Or `StyledComponents.js`, actually.

Comment (jorgeorpinel) [reply src/Community/Block/index.js]: or even `comps/styled.js` :p

**Comment by jorgeorpinel** [comment]
Before doing a full review here, is there any context of what this community page is for? Motivation, etc? Thanks

**Comment by shcheklein** [comment]
@jorgeorpinel motivation, etc - not a question for @iAdramelk, we can discuss this with @SvetaGr next all hands, for example, or in Slack.

**Comment by jorgeorpinel** [comment]
From Slack conversation, here's the...

Purpose:
- Provide an overview of what's going on with the project on the community side.
- For everyone who is meeting DVC for the first time.
- Help ppl find all the resources that might be useful to community members in one place.
- Emphasize that we are supporting "ambasssadors" among community members: ppl who give talks, want to organize meetup, etc.
- Remove some of the scattered mentions/links to community resources – which exact ones? We'll see.

Motivation:
- Most of the community page content was already there but scattered over the website and sometimes hard to find: "resources, channels of communication being pretty disorganized and mentioned both in the top menu, in the "support" section, in the "help" section, etc."

**Comment by jorgeorpinel** [comment]
Just looking at https://dvc-landing-community-p-ins0im.herokuapp.com/community (no full code review from me yet):

- Is removing the GitHub link from the top nav a good idea?
- The hero images of each section move to the bottom in responsive view, maybe they should stay on top behind the title of the section (e.g. the picture in Cleveland for **Meet the Community)**
- The subsections are really great but in responsive view I feel like they become too big and it takes a long time to scroll through them. Maybe they can have smaller font or less height or even hide some from that view?
- The **Be an Ambassador** block may need more emphasis, maybe a special color and/or being on top of its section.
- The **Documentation** block may not be necessary? Since you're already in the docs website. Perhaps at least move to the last one in regular view, and hide from responsive.
- Have we considered a little ToC sidebar or floating widget to jump from section to section? For example so you see that there's events without having to scroll all the way down (and clicking on a **Events** link takes you to the bottom)

File: src/Community/Block/index.js (hunk 2 above)
Comment (iAdramelk) [reply src/Community/Block/index.js]: Yep, functional components are a lot easier to read for me, and hooks make using state and side effects a lot easier and cleaner.

File: src/Community/Block/index.js (hunk 1 above)
Comment (iAdramelk) [reply src/Community/Block/index.js]: I'm ok either way, as long as they are all named the same and are in the same folder os the main component.

File: src/Community/Block/index.js (hunk 2 above)
Comment (jorgeorpinel) [reply src/Community/Block/index.js]: Thanks. Will keep this in mind.

File: src/Community/Block/index.js (hunk 1 above)
Comment (jorgeorpinel) [reply src/Community/Block/index.js]: OK, I vote for `comps/styled.js`. Seems like the shortest, cleanest option to me.

**Comment by SvetaGr** [comment]
Thanks for your feedback, @jorgeorpinel, those are great comments! Some answers after discussion with @shcheklein: 
- the mobile version has a lot to be done and most of the feedback you have up here  coincide with what already been discussed with evil martians by @shcheklein. It's a work in progress. 
- The menu is still under discussion and we are not happy with current version tool. WiP again. 
- The be an Ambassador block is positioned at the end because it seems like the logic end of the funnel for us., e.g. for now to be an ambassador, you have to start with being familiar with DVC thus Make a PR goes first first. It can surely change with time. 
- The Documentation block is mentioned as one of the ways to Learn more and learn from the community as well (e.g. user-generated tutorials). 
- flowing widget is a great idea, we should add it to the next iteration for the page.

**Comment by SvetaGr** [comment]
https://dvc-landing-community-p-iasedq.herokuapp.com/community New working link

**Comment by andronovhopf** [comment]
Some minor copy-edits!

**Under "Contribute"**
`DVC is 100% open-source and we welcome contributions of all sorts to our code and docs.` change to 
"DVC is 100% open source. We welcome contributions to our codes and docs."

`Become DVC contributor and let us build something great together.` change to "Let's build something great together. Become a DVC contributor!"

`Have something cool on your mind? Suggest a text and we'll publish it in our blog.` change to
"We're always interested in guest writers for our blog. If you've got something to share, please reach out."

`Get perks and benefits for significant contributions, creating content or hosting meetups.` change to
"Get perks and benefits for contributions like creating content or hosting meetups."

**Under "Learn"**
`Every month we are sharing here our news, findings, interesting reads, community takeaways, and everything along the way.` change to "Our community is a place for beginners and pros alike to learn best practices for data science and engineering. Check out our documentation, news from the community, tutorials and more."

**Under "Events"**
`Conferences, meetups and talks to learn more about DVC or meet in person.` change to "Learn more about DVC and meet us in person! We're appearing at conferences, meetups, and talks around the world."

File: src/Community/Block/index.js (hunk 1 above)
Comment (iAdramelk) [reply src/Community/Block/index.js]: One more consideration: We can replace styled components with something else in the future, so maybe we shouldn't make styles file name based on them.

File: src/Community/data.json [hunk 3]
```diff
@@ -0,0 +1,78 @@
+{
+  "hero": {
+    "picture": "/static/img/community/hero.png",
+    "url": "http://event.link"
+  },
+  "section": {
+    "contribute": {
+      "title": "Contribute",
+      "description": "DVC is 100% open-source and we welcome contributions of all sorts to our code and docs."
```
Comment (jorgeorpinel) [inline src/Community/data.json]:
```suggestion
      "description": "DVC is 100% open source. We welcome contributions to our code and docs."
```
per https://github.com/iterative/dvc.org/pull/874#issuecomment-571841464

Comment (jorgeorpinel) [reply src/Community/data.json]: p.s. @iAdramelk is the pre-commit hook not asking you to rewrap these lines? I see Restyled has some complaints BTW (but not this one). Are you using `git commit --no-verify`?

File: src/Community/Contribute/index.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+          <Item>
+            <CommunityBlock
+              title="Make a PR"
+              action={
+                <CommunityButton theme={theme} href="/">
+                  Open Chat
+                </CommunityButton>
+              }
+            >
+              Become DVC contributor and let us build something great together.
```
Comment (jorgeorpinel) [inline src/Community/Contribute/index.js]:
```suggestion
              Let's build something great together. Become a DVC contributor!
```

File: src/Community/Contribute/index.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            <CommunityBlock
+              title="Write a blogpost"
+              action={
+                <CommunityButton theme={theme} href="/">
+                  Learn More
+                </CommunityButton>
+              }
+            >
+              Have something cool on your mind? Suggest a text and we&apos;ll
+              publish it in our blog.
```
Comment (jorgeorpinel) [inline src/Community/Contribute/index.js]: We're always interested in guest writers for our blog. If you've got something to share, please reach out.

File: src/Community/Contribute/index.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+            <CommunityBlock
+              title="Be an Ambassador"
+              action={
+                <CommunityButton theme={theme} href="/">
+                  Apply
+                </CommunityButton>
+              }
+            >
+              Get perks and benefits for significant contributions, creating
+              content or hosting meetups.
```
Comment (jorgeorpinel) [inline src/Community/Contribute/index.js]: Get perks and benefits for contributing to the code base, writing blogs, hosting meetups and more.

File: src/Community/data.json [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    },
+    "events": {
+      "title": "Events",
+      "description": "Conferences, meetups and talks to learn more about DVC or meet in person."
+    },
+    "learn": {
+      "title": "Learn",
+      "description": "Every month we are sharing here our news, findings, interesting reads, community takeaways, and everything along the way."
```
Comment (jorgeorpinel) [inline src/Community/data.json]:
```suggestion
      "description": "Our community is a place for beginners and pros alike to learn best practices for data science and engineering. Check out our documentation, news from the community, tutorials and more."
```

File: src/Community/data.json [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    "url": "http://event.link"
+  },
+  "section": {
+    "contribute": {
+      "title": "Contribute",
+      "description": "DVC is 100% open-source and we welcome contributions of all sorts to our code and docs."
+    },
+    "events": {
+      "title": "Events",
+      "description": "Conferences, meetups and talks to learn more about DVC or meet in person."
```
Comment (jorgeorpinel) [inline src/Community/data.json]:
```suggestion
      "description": "Learn more about DVC and meet us in person! We're appearing at conferences, meetups, and talks around the world."
```

File: src/Community/Contribute/index.js (hunk 6 above)
Comment (jorgeorpinel) [reply src/Community/Contribute/index.js]: Not sure about this "and more" ending here. Maybe "or other community building activities" @andronovhopf?

File: src/Community/data.json [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    "meet": {
+      "title": "Meet the Community",
+      "description": "Ask questions, find answers, share your ideas and connect with DVC community."
+    }
+  },
+  "documentation": [
+    {
+      "url": "/doc/get-started",
+      "title": "Get started",
+      "description": "Learn how to set up your project"
```
Comment (jorgeorpinel) [inline src/Community/data.json]:
```suggestion
      "description": "Step-by-step introduction into basic DVC features"
```

File: src/Community/data.json [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+    },
+    {
+      "url": "/doc/tutorials",
+      "title": "Tutorials",
+      "description": "Learn how to solve most typical tasks"
```
Comment (jorgeorpinel) [inline src/Community/data.json]:
```suggestion
      "description": "Reproducible walkthroughs on advanced topics"
```

File: src/Community/data.json [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+    },
+    {
+      "url": "/doc/changelog",
+      "title": "Changelog",
+      "description": "See what's new in DVC core "
```
Comment (jorgeorpinel) [inline src/Community/data.json]:
```suggestion
      "description": "See what's new in DVC."
```

Comment (jorgeorpinel) [reply src/Community/data.json]: p.s. we're probably going to remove this change log soonish. See #897.

File: src/Community/data.json [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+      "title": "Switching to DVC with no pain",
+      "author": "Lucy Yii",
+      "date": 1576072266986
+    }
+  ],
+  "events": [
+    {
+      "url": "/1",
+      "title": "Big Data Conf'19",
+      "description": "DVC team will give a talk about deployment DVC environment on Heroku server",
```
Comment (jorgeorpinel) [inline src/Community/data.json]:
```suggestion
      "description": "Our talk about deployment DVC environment on Heroku",
```

File: src/Community/data.json [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+      "description": "See what's new in DVC core "
+    }
+  ],
+  "userContent": [
+    {
+      "url": "/1",
+      "title": "How we use DVC in our project",
+      "author": "James Hopfield",
+      "date": 1576072266986
+    },
```
Comment (jorgeorpinel) [inline src/Community/data.json]: Can we use human readable dates instead of timestamps for these? e.g. `yyyy-mm-dd` format

File: src/Community/Block/index.js (hunk 1 above)
Comment (jorgeorpinel) [reply src/Community/Block/index.js]: BTW I take that back. I just realized these files are already inside component named directories so I'd just rename them to `styled.js` (no further `comps/` dir).
About replacing this later, that's OK, we can rename the files again if needed. Who knows how the replacement will look, maybe these files will be totally gone anyway.

File: src/Community/Contribute/index.js (hunk 6 above)
Comment (andronovhopf) [reply src/Community/Contribute/index.js]: @jorgeorpinel I think "or other community building activities" is too wordy. If we don't like "and more", we could cut it entirely so it reads `Get perks and benefits for contributing to the code base, writing blogs, and hosting meetups.`

Comment (jorgeorpinel) [reply src/Community/Contribute/index.js]: How about "writing blogs, hosting meetups, etc." ? 🙂

Comment (andronovhopf) [reply src/Community/Contribute/index.js]: Personally, I prefer "and more" to "etc." in blurbs- but that's my stylistic preference. Either way, it's readable and makes sense. No problem if you go ahead with that!

Comment (jorgeorpinel) [reply src/Community/Contribute/index.js]: Yeah I don't love "etc." either. I guess we can just end the sentence there like you mentioned. "and more" sounds to me like something we're offering, as if this was a list of perks and not of what we wish our ambassadors to help with.

Ok, updated the suggestion above!

File: src/Community/data.json (hunk 13 above)
Comment (iAdramelk) [reply src/Community/data.json]: We can use any format there, that can be parsed by `new Date()`, so `yyyy-mm-dd` should work too.

File: src/Community/data.json (hunk 3 above)
Comment (iAdramelk) [reply src/Community/data.json]: Nope. I think the problem is that restyled didn't read `.prettierrc` and is using it's own defaults to validate. I think that this is fixed in master, but branch is behind this change.

File: src/Community/data.json (hunk 13 above)
Comment (jorgeorpinel) [reply src/Community/data.json]: Great, so do you want to update this to the corresponding date in readable format? 🙂

File: src/Community/data.json (hunk 3 above)
Comment (jorgeorpinel) [reply src/Community/data.json]: Ah OK. Yeah so feel free to merge from `master` any time. Thanks!

File: src/Community/data.json (hunk 13 above)
Comment (iAdramelk) [reply src/Community/data.json]: It's just a placeholder data now, I think to replace it then we will add actual events.

**Comment by shcheklein** [comment]
@iAdramelk please, merge master, resolve conflicts

File: pages/api/blog.js [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+      res.status(502).json({ error: 'Unexpected response from Blog' })
+
+      return
+    }
+
+    const data = await response.text()
+
+    res.status(200).json(data)
+  } catch {
+    res.status(404)
```
Comment (fabiosantoscode) [inline pages/api/blog.js]: Is this the appropriate status code? Maybe 503/502 is better since it's an upstream error.

File: pages/community.js [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+      href="https://cdnjs.cloudflare.com/ajax/libs/slick-carousel/1.6.0/slick.min.css"
+    />
+    <link
+      rel="stylesheet"
+      type="text/css"
+      href="https://cdnjs.cloudflare.com/ajax/libs/slick-carousel/1.6.0/slick-theme.min.css"
+    />
+    <title>Community | {META_BASE_TITLE}</title>
+  </Head>
+)
```
Comment (fabiosantoscode) [inline pages/community.js]: nitpick: this could be a `head` and used below as `{head}` since it's static.

Don't feel obligated to change this if you feel like this is a better solution.

File: src/components/Community/Button/index.js [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+    backgroundColor: PropTypes.string,
+    color: PropTypes.string
+  }),
+  forwardedRef: PropTypes.func
+}
+
+// eslint-disable-next-line react/display-name
+export default React.forwardRef((props, ref) => (
+  <BareButton {...props} forwardedRef={ref} />
+))
```
Comment (fabiosantoscode) [inline src/components/Community/Button/index.js]: I think you can make the BareButton a wrappee of `React.forwardRef` itself, instead of adding an extra component around it. It should work fine even if you add hooks to the component eventually.

File: src/components/Community/Contribute/index.js [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+            </CommunityBlock>
+          </Item>
+          <Item>
+            <CommunityBlock
+              title="Write a blogpost"
+              action={
+                <CommunityButton
+                  theme={theme}
+                  href="mailto:info@dvc.org?subject=I want to write a blogpost!"
+                  target="_blank"
```
Comment (fabiosantoscode) [inline src/components/Community/Contribute/index.js]: nitpick: the only way of making mailto links open in a new tab is using `window.open`.

File: pages/community.js (hunk 15 above)
Comment (iAdramelk) [reply pages/community.js]: I actually moved it inside the component on the other pages already, so probably I'll do it here too.

File: pages/api/blog.js (hunk 14 above)
Comment (iAdramelk) [reply pages/api/blog.js]: You are right, will change.

File: src/components/Community/Button/index.js (hunk 16 above)
Comment (iAdramelk) [reply src/components/Community/Button/index.js]: It's mostly done this way to let the button have a readable name to show in dev tools, bow I start to think that I can do both without separate component, but I need to experiment some more.

File: src/components/Community/Contribute/index.js (hunk 17 above)
Comment (iAdramelk) [reply src/components/Community/Contribute/index.js]: It's ok not to open new page if the user use external email app. We need it only for browser email client like gmail.

@shcheklein can you check if the links are now opened in the new tab in your case please?

File: pages/api/blog.js (hunk 14 above)
Comment (iAdramelk) [reply pages/api/blog.js]: You are right, will change it.

File: pages/community.js (hunk 15 above)
Comment (iAdramelk) [reply pages/community.js]: I already moved it inside the component for other pages, so I probably will do it here too.

File: src/components/Community/Button/index.js (hunk 16 above)
Comment (iAdramelk) [reply src/components/Community/Button/index.js]: Tried, didn't work (you can't default export consts, only functions). But I renamed BareButton to CommunityButton for consistency.

File: src/Community/data.json (hunk 3 above)
Comment (jorgeorpinel) [reply src/Community/data.json]: Suggestion still pending 🙂

File: src/Community/Contribute/index.js (hunk 4 above)
Comment (jorgeorpinel) [reply src/Community/Contribute/index.js]: Pending

File: src/Community/Contribute/index.js (hunk 5 above)
Comment (jorgeorpinel) [reply src/Community/Contribute/index.js]: Same...

File: src/Community/data.json (hunk 9 above)
Comment (jorgeorpinel) [reply src/Community/data.json]: This whole review also seems pending.

**Comment by shcheklein** [comment]
@iAdramelk looks like a lot of good reviews and suggestions were lost due to a lot of changes. @SvetaGr @jorgeorpinel @jorgeorpinel could you summarize in one page, please, what copy updates do you suggest to make? 

@jorgeorpinel if you have anything related to code - could you please make a separate issue with checkboxes to discuss and fix them?

Thanks!

**Comment by jorgeorpinel** [comment]
I'll just apply the copy edits in a regular update PR soon, so don't worry about my https://github.com/iterative/dvc.org/pull/874#pullrequestreview-357831784 above Alex.

Sorry, I didn't really get a chance to review the JS code here Ivan, my review on `src/Community/data.json` is also on content (will include in my regular updates ASAP).

Thanks

**Comment by shcheklein** [comment]
@jorgeorpinel in this specific case, let's create a list of changes please since @andronovhopf and @SvetaGr had some changes in mind as well. I would like to consolidate this and do at once.

**Comment by jorgeorpinel** [comment]
OK let's use #996
